### ISSUE TITLE
Monitoring with Prometheus (and metrics-server)

### ISSUE THREAD

**Issue by bradenmacdonald** [openedx/openedx-k8s-harmony#3]
Use helm subcharts to deploy Prometheus and Grafana onto the chart.

TBD: Can we have them auto-detect and start monitoring Open edX instances on the cluster as they get deployed using Tutor, like how the ingress controller works?

**Comment by antoviaque** [comment]
@bradenmacdonald Thanks for creating this task! 

@lpm0073 was it the one you are taking on?

**Comment by lpm0073** [comment]
yes, this is me

**Comment by antoviaque** [comment]
@lpm0073 Alright, I'm assigning the issue to you then, if that works :)

**Comment by antoviaque** [comment]
@lpm0073 Could you post a status update on this task here? This way we could follow & discuss here async, ahead of the next meeting.

**Comment by antoviaque** [comment]
@lpm0073 Are you still interested in this task?

**Comment by antoviaque** [comment]
Recap from the meeting: now @lpm0073 is unblocked to work on this ticket, based if I understood correctly on the autoscaling work from @jfavellar90 in https://github.com/openedx/tutor-contrib-multi/issues/2

**Comment by felipemontoya** [comment]
@lpm0073 could you give us an update on this? are you interested in pursuing this still?

**Comment by bradenmacdonald** [comment]
@lpm0073 Thanks! Note that metrics-server and VPA are already being worked on in #17

**Comment by lpm0073** [comment]
Question to the group: aside from the helm charts, there are a few AWS resources that need to exist, and need to be provided to the helm chart:

- IAM role for service accounts
- EC2 instance profile
- EC2 tagging role and policy attachment

I have [these Terraform scripts](https://github.com/lpm0073/cookiecutter-openedx-devops/blob/main/%7B%7Bcookiecutter.github_repo_name%7D%7D/terraform/stacks/modules/kubernetes_prometheus/main.tf). how should i incorporate these into this repo?

**Comment by bradenmacdonald** [comment]
@lpm0073 I put some DigitalOcean example Terraform Scripts in https://github.com/openedx/openedx-k8s-harmony/tree/main/infra-example ; you could rename that folder to infra-example-do and create a new infra-example-aws folder with the AWS terraform.

**Comment by lpm0073** [comment]
confirming that [pr #17](https://github.com/openedx/openedx-k8s-harmony/pull/17) takes care of metrics-server and vpa dependencies for this issue.

**Comment by felipemontoya** [comment]
After PR#17 is merged we will need to create a new PR with the helm charts for grafana and prometheus

**Comment by antoviaque** [comment]
@adzuci Did you get the information and discussions you wanted during the last meeting about this?

Note that @bradenmacdonald has also created a dedicated task to follow-up on topics of interest for you and 2U, at https://github.com/openedx/openedx-k8s-harmony/issues/28 - comments there are welcomed! Or during the meeting later today.

**Comment by felipemontoya** [comment]
@lpm0073 we are getting close to finisht the autoscaling part of the charts which were closely related to this.

Are you interested in pursuing this further?

**Comment by felipemontoya** [comment]
@lpm0073 following the meeting we will split this ticket into two less ambiguous issues. Please comment if you have a different opinion

**Comment by felipemontoya** [comment]
Now that the issues for grafana and prometheus have been created we can go ahead and close this.
